### ISSUE TITLE
Timestamps across the web site / make format user selectable

### ISSUE THREAD

**Issue by woodpeck** [openstreetmap/openstreetmap-website#631]
This is a feature request for user-selectable date/time styles across the web site. A detailed explanation follows below.

For a long time we've been using vague timestamps like "about 5 years ago" in several places in the UI. - I see where the motivation comes from (less clutter, nicer on the eye, easier to translate, no time zone issues) but I've always hated it.

The reason is that more often than not, I was using the web site to investigate something. Was a certain edit _before_ or _after_ a certain other edit? Did they make that change _after_ I sent them the message? Was this account created before or after I blocked that other account? Was the block they got a reason for the edit, or was the edit the reason for the block? And so on. Precise time stamps make this kind of work much easier, especially if you're looking at a time ranger where everything is "about one year ago".

With the recent UI change, these imprecise time stamps are used in even more places, occasionally augmented by a hover facility that will show a precise time stamp. In this trac ticket https://trac.openstreetmap.org/ticket/5045 the issue has been discussed, and the complainant has made available a hack that would allow people to have precise time stamps at least where the hover code is enabled. 

So far, so good, and you'll say that my use case is niche and anyway I could build tools for myself to find out what I need to know. BUT, and I say this from my DWG background: Going forward, we will need more "ordinary users" taking stewardship if data, investigating strange edits and wrongdoings, reporting or flagging bad ones, even dealing with them themselves. The "niche" that DWG currently occupies will have to widen in the future, with more and more people looking at the finer details of what is going on in OSM. 

While "5 hours ago" is nice for the casually browsing user, it makes life harder for those who actually want to do some sleuthing, and we don't want to make it harder for them.

Yes it would be terrific if we could have a second web site that was specifically aimed at experts but we won't any time soon. So in the mean time it would be great if the needs of more analytically inclined users were kept in mind when changing things. It would be great if, in their profile settings, users could choose whether they want precise timestamps throughout, or the reader-friendly "5 hours ago" variant.

**Comment by SomeoneElseOSM** [comment]
(re the comments above about tooltips) they don't appear everywhere.  See for example:

http://www.openstreetmap.org/user_blocks/493

**Comment by HolgerJeromin** [comment]
Just another topic are mobile browsers where you have no way to get the exact Information.

**Comment by polarbearing** [comment]
I fully support @woodpeck's arguments, as an "ordinary user" checking changes in my neighbourhood. The current vague timestamps make my life difficult as well, along with the lack of clear page numbers for a user's history, as in #1089. As said there, we should have at least the option to switch to classic timestamps and pagination, with page one on the oldest changesets, as it is typical in forums, of course showing the latest first.

**Comment by HolgerJeromin** [comment]
We could implement somthing like
https://bitsofco.de/making-abbr-work-for-touchscreen-keyboard-mouse/
But moving to `data-title` to prevent duplicate tooltip on devices with a mouse.

**Comment by boothym** [comment]
Yes please, let's implement the CSS tooltip mentioned above. It's a bit annoying if you want to know the exact timestamp when using your phone.

**Comment by lgommans** [comment]
Been meaning to find the right file to commit this into and propose it as pull request but I keep not getting around to it...

Simple CSS-only solution that doesn't change anything else except lets you see the title text for `abbr` elements on mobile and desktop consistently:

![screenshot of a webpage, the URL bar shows test.html and the page contains a sentence with a tooltip in the middle, triggered (in this case) by positioning the mouse over it](https://github.com/user-attachments/assets/a38b02f5-216c-4de9-a995-4f6f1f5415f4)

(The cursor is actually a hand/pointer; screenshot software does not capture that.)

```html
<style>
    abbr[title] {
        position: relative;

        /* ensure consistent styling across browsers */
        text-decoration: underline dotted;
    }

    abbr[title]:hover::after,
    abbr[title]:focus::after {
        content: attr(title);

        /* position tooltip like the native one */
        position: absolute;
        left: 0;
        bottom: -30px;
        width: auto;
        white-space: nowrap;

        /* style tooltip */
        background-color: #1e1e1e;
        color: #fff;
        border-radius: 3px;
        box-shadow: 1px 1px 5px 0 rgba(0, 0, 0, 0.4);
        font-size: 14px;
        padding: 3px 5px;
    }
</style>
<a href="/issues/">There are <abbr title="seven!" tabindex=0>how many?</abbr> issues with this approach!</a>
```

CSS from <https://stackoverflow.com/a/69353021/1201863>. Tested that the code works in normal text and also nested, for example in an `<a>` link

The `tabindex` makes it selectable such that screen reader users can hopefully also open it. Regarding the value, [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) `tabindex=0` "means that the element should be focusable in sequential keyboard navigation" -- it does *not* put it at the top of the tab stack!

Hopefully this covers everyone's (accessibility) requirements by looking like what people expect from a tooltip while also not requiring javascript. I'm kind of blown away that mobile browsers, after all these years, still can't just do tooltips natively...
